runtime/JavaScript/src/antlr4/Token.d.ts

```diff
@@ -7,6 +9,7 @@ export declare class Token {
     static DEFAULT_CHANNEL: number;
     static HIDDEN_CHANNEL: number;
 
+    source: [ TokenSource, InputStream ];
```

Comment:
I'd rather you add `getTokenSource` and `getInputStream`. `source` being a tuple is an internal detail

Reply:
I have a method where I refer to the source property and a getTokenSource() would look strange in my opinion:.
```
private getCommonTokenByToken(oldToken: Token): CommonToken {
    const cToken = new CommonToken(oldToken.source, oldToken.type, oldToken.channel, oldToken.start, oldToken.stop);
    cToken.tokenIndex = oldToken.tokenIndex;
    cToken.line = oldToken.line;
    cToken.column = oldToken.column;
    cToken.text = oldToken.text;
    return cToken;
}
```

Reply:
Sorry, maybe what I wrote was a little misunderstood.
Perhaps it would have been more understandable if I had referred to a public field instead of a property.
However, VS Code [marks](https://putridparrot.com/blog/properties-and-fields-in-typescript/) public fields as properties.

Reply:
I'm really not keen to expose such a poorly designed construct. The d.ts file already has `getInputStream`, so you only need to add `getTokenSource`.